src/v/kafka/server/protocol.h

```diff
@@ -135,6 +141,7 @@ class protocol final : public rpc::server::protocol {
     ss::sharded<security::credential_store>& _credentials;
     ss::sharded<security::authorizer>& _authorizer;
     ss::sharded<cluster::security_frontend>& _security_frontend;
+    cluster::data_policy_manager& _data_policy_manager;
```

Comment:
I think it is better to start off with a sharded type here. It looks safe now, but I can't imagine that it will be able to remain non-sharded for very long.

Specifically I would keep the data policy manager outside the cluster namespace, and then created a sharded<data policy manager> (notice this would then mean that the hash table is no longer sharded) in the application and pass it into the data policy front end. then pass both into the kafka layer. this is the same pattern that cluster::security_frontend and security::authorizer uses.